test/modules/XMLHttpRequest/response/xhr-response-without-body.test.ts

```diff
@@ -0,0 +1,108 @@
+// @vitest-environment jsdom
+import { afterAll, afterEach, beforeAll, expect, it, vi } from 'vitest'
+import { HttpServer } from '@open-draft/test-server/http'
+import { XMLHttpRequestInterceptor } from '../../../../src/interceptors/XMLHttpRequest'
+import { createXMLHttpRequest, useCors } from '../../../helpers'
+import type { HttpRequestEventMap } from '../../../../src'
+
+const httpServer = new HttpServer((app) => {
+  app.use(useCors)
+  app.get('/:statusCode', (req, res) =>
+    res.status(+req.params.statusCode).end()
+  )
+})
+
+const interceptor = new XMLHttpRequestInterceptor()
+
+const responseListener = vi.fn<HttpRequestEventMap['response']>()
+interceptor.on('response', responseListener)
+
+beforeAll(async () => {
+  await httpServer.listen()
+  interceptor.apply()
+})
+
+afterEach(() => {
+  vi.resetAllMocks()
+})
+
+afterAll(async () => {
+  interceptor.dispose()
+  await httpServer.close()
+})
+
+it('represents a 204 response without body using fetch api response', async () => {
```

Comment:
Take a look, I've expanded the tests, creating three separate test cases instead of iterating over the list of status codes. It's crucial to keep the tests as flat and straightforward as possible. This means as little logic specific to tests as possible, like `test.each()`. This results in tests that are easy to find, read, change, and remove.